app/controllers/wishlists_controller.rb

```diff
@@ -3,50 +3,57 @@ class WishlistsController < ApplicationController
 
   # GET /wishlists or /wishlists.json
   def index
-    @wishlists = policy_scope(Wishlist)
+    @wishlists = current_user.wishlists.includes(:activity)
   end
 
   # GET /wishlists/1 or /wishlists/1.json
   def show
-    authorize @wishlist
+    # authorize @wishlist
   end
 
   # GET /wishlists/new
   def new
     @wishlist = Wishlist.new
-    authorize @wishlist
+    # authorize @wishlist
   end
 
   # GET /wishlists/1/edit
   def edit
-    authorize @wishlist
+    # authorize @wishlist
   end
 
   # POST /wishlists or /wishlists.json
   def create
-    @wishlist = Wishlist.new(wishlist_params)
-    @wishlist.user = current_user
-    authorize @wishlist
+    @activity = Activity.find_by_id(params[:activity_id])
+    @wishlist = Wishlist.find_by(activity: @activity, user: current_user)
+
```

Comment:
oh this is where YC meant in this slack thread 
https://lewagon-alumni.slack.com/archives/C03P9ERQX6V/p1663261898545369